odoo/addons/base/models/ir_http.py

```diff
@@ -137,25 +137,34 @@ class IrHttp(models.AbstractModel):
     _description = "HTTP Routing"
 
     @classmethod
-    def _slugify_one(cls, value: str, max_length: int = 0) -> str:
+    def _slugify_one(cls, value: str, max_length: int = None) -> str:
         """ Transform a string to a slug that can be used in a url path.
             This method will first try to do the job with python-slugify if present.
-            Otherwise it will process string by stripping leading and ending spaces,
-            converting unicode chars to ascii, lowering all chars and replacing spaces
-            and underscore with hyphen "-".
+            Otherwise it will process string by replacing spaces and underscores with
+            dashes '-',removing every character that is not a word or a dash,
+            collapsing multiple dashes like --- into a single dash, removing leading
+            and trailing dashes and converting to lowercase.
+            Example: ^h☺e$#!l(%l}o 你好& becomes hello-你好
         """
         if slugify_lib:
             # There are 2 different libraries only python-slugify is supported
             try:
                 return slugify_lib.slugify(value, max_length=max_length)
             except TypeError:
                 pass
-        uni = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore').decode('ascii')
-        slug_str = re.sub(r'[\W_]+', '-', uni).strip('-').lower()
-        return slug_str[:max_length] if max_length > 0 else slug_str
+        uni = unicodedata.normalize('NFKD', value)
+        slugified_segments = []
+        for slug in re.split('-|_| ', uni):
+            slug = re.sub(r'([^\w-])+', '', slug)
+            slug = re.sub(r'--+', '-', slug)
+            slug = slug.strip('-')
+            if slug:
+                slugified_segments.append(slug.lower())
+        slugified_str = ('-').join(slugified_segments)
```

Comment:
Nitpicking, it is uncommon to put the parenthesis around a literal string. You only do it on literal int because otherwise the parser thinks it is a float.
```suggestion
        slugified_str = '-'.join(slugified_segments)
```